src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -125,6 +132,102 @@ private Expression ParseExpressionSegment()
 			}
 		}
 
+		// => operator
+		private Expression ParseLambdaExpression()
+		{
+			// in case the expression is not a lambda, we have to restart parsing
+			var originalPos = _token.pos;
+
+			try
+			{
+				var parameters = ParseLambdaParameterList();
+				ValidateToken(TokenId.LambdaArrow);
+
+				var startExpr = _parsePosition;
+
+				var parenCount = 0;
+				var inLambdaBody = true;
+				while (inLambdaBody)
+				{
+					NextToken();
+					if (_token.id == TokenId.End)
+						inLambdaBody = false;
+					if (_token.id == TokenId.OpenParen)
+						parenCount++;
+					if (_token.id == TokenId.CloseParen)
+						parenCount--;
+
+					// lambda is a function parameter
+					if (parenCount == 0 && _token.id == TokenId.Comma)
+						inLambdaBody = false;
+
+					// body closure
+					if (parenCount < 0)
+						inLambdaBody = false;
+				}
+
+				var lambdaBodyExp = _expressionText.Substring(startExpr, _token.pos - startExpr);
+				return new InterpreterExpression(_arguments.Settings, lambdaBodyExp, parameters);
+			}
+			catch (Exception)
```

Comment:
Can we reduce the exception scope to only catch `ParseException` type?

Reply:
Catching `ParseException` seems to work (all unit tests pass when they're parsed with the lambda expressions active). My original idea was to always fallback to the existing parsing in case the expression can't be parsed as a lambda expression, which is why I caught `Exception`. 

I'll change it to only catch `ParseException`; if there's an issue we can always change it back later :)